test/test_cpp20_no_sourcelocation/custom_error.cpp

```diff
@@ -0,0 +1,68 @@
+#include "pch.h"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+
+namespace
+{
+    static bool s_loggerCalled = false;
+
+    // Note that we are checking that the source line number matches expectations.  If lines above this are changed
+    // then this value needs to change as well.
+    void FailOnLine15()
+    {
+        // Validate that handler translated exception
+        REQUIRE_THROWS_AS(check_hresult(0x80000018), hresult_illegal_delegate_assignment);
+    }
+
+    static struct {
+        uint32_t lineNumber;
+        char const* fileName;
+        char const* functionName;
+        void* returnAddress;
+        winrt::hresult result;
+    } s_loggerArgs{};
+
+    void __stdcall logger(uint32_t lineNumber, char const* fileName, char const* functionName, void* returnAddress, winrt::hresult const result) noexcept
+    {
+        s_loggerArgs = {
+            .lineNumber = lineNumber,
+            .fileName = fileName,
+            .functionName = functionName,
+            .returnAddress = returnAddress,
+            .result = result,
+        };
+        s_loggerCalled = true;
+    }
+}
+
+#if defined(__clang__)
+// FIXME: Blocked on __cpp_consteval, see:
+// * https://github.com/microsoft/cppwinrt/pull/1203#issuecomment-1279764927
+// * https://github.com/llvm/llvm-project/issues/57094
+TEST_CASE("custom_error_logger", "[!shouldfail]")
+#else
+TEST_CASE("custom_error_logger")
+#endif
```

Comment:
Ah, the failure on all Clang builds (including clang-cl) is caused by this "[!shouldfail]" tag. This whole `#if defined(__clang__)` condition should be removed because the lack of source location support on Clang does not matter on this test, which disables source location explicitly.

Reply:
That makes sense.  I'll try that.